backend/src/main/kotlin/me/andrewda/handlers/PaymentHandler.kt

```diff
@@ -0,0 +1,76 @@
+package me.andrewda.handlers
+
+import io.ktor.application.call
+import io.ktor.auth.authenticate
+import io.ktor.request.receiveOrNull
+import io.ktor.response.respondRedirect
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.post
+import io.ktor.routing.route
+import me.andrewda.controllers.PaymentController
+import me.andrewda.controllers.PersonController
+import me.andrewda.controllers.RequestController
+import me.andrewda.models.NewPayment
+import me.andrewda.payment.PayPal
+import me.andrewda.utils.*
+
+fun Route.payment() {
+    route("/payments") {
+        authenticate {
+            route("/people") {
+                post("/{slug}") {
+                    val slug = call.parameters["slug"] ?: throw NotFound()
+                    val newPayment = call.receiveOrNull<NewPayment>() ?: throw MissingFields()
+                    val user = call.getUser() ?: throw NotAuthenticated()
```

Comment:
Should we make it mandatory that a donator needs to have an account? I thought we were going to allow non-members to donate.

Reply:
Good point